File: CMOWebApi.Models/TenderPressRelease/PressReleaseModel.cs
using CMOWebApi.Models.GeneralModel;
using System;
using System.Collections.Generic;

namespace CMOWebApi.Models.VendorPressRelease
{
    public class PressReleaseModel
    {
        public long Id { get; set; }
        public Nullable<long> Code { get; set; }
        public Nullable<int> DepartmentCode { get; set; }
        public Nullable<int> CategoryCode { get; set; }
        public Nullable<int> SubCategoryCode { get; set; }
        public Nullable<int> DistrictCode { get; set; }
        public string GeneralDescription { get; set; }
        public string Description { get; set; }
        public string URL { get; set; }
        public Nullable<int> DisplayOrder { get; set; }
        public string PDFUrl { get; set; }
        public List<string> PDFUrlList { get; set; }
        public List<string> ImageUrlList { get; set; }
        public string ImageUrl { get; set; }
        public string HomePageImageUrl { get; set; }
        public string KeyWords { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public System.DateTime CreatedDate { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public bool IsActive { get; set; } = true;
        public bool IsDeleted { get; set; } = false;
        public List<string> VIPCategoryList { get; set; }
        public List<string> VIPPersionList { get; set; }
        public Nullable<bool> IsSpecialPressRelease { get; set; }
        public Nullable<long> PressReleaseLevelCode { get; set; }
        public Nullable<System.DateTime> PressreleaseDate { get; set; }
        public List<string> PressReleaseDepartmentMappingList { get; set; }
        public List<string> DistrictList { get; set; }

        public Nullable<bool> IsOldRecord { get; set; }
        public string PressReleaseTime { get; set; }
        public string DistrictName { get; set; }
        public string VIPPersonName { get; set; }
        public Nullable<decimal> DIPR_Id { get; set; }
        public Nullable<long> LookupCategoryCode { get; set; }
        public string NameOfVIPPerson { get; set; }
        public string AmountinLakh { get; set; }
        public string NoOfInaugration { get; set; }
        public string NoOfLokarpan { get; set; }
        public string NoOfNewInitatives { get; set; }
        public string Latitude { get; set; }
        public string Longitude { get; set; }
    }

    public class PressReleaseListModel
    {
        public long Id { get; set; }
        public Nullable<long> Code { get; set; }
        public Nullable<int> DepartmentCode { get; set; }
        public Nullable<int> CategoryCode { get; set; }
        public Nullable<int> SubCategoryCode { get; set; }
        public Nullable<int> DistrictCode { get; set; }
        public string GeneralDescription { get; set; }
        public string Description { get; set; }
        public string URL { get; set; }
        public Nullable<long> DisplayOrder { get; set; }
        public string PDFUrl { get; set; }
        public string ImageUrl { get; set; }
        public string HomePageImageUrl { get; set; }
        public string KeyWords { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public System.DateTime CreatedDate { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public Nullable<decimal> DIPR_Id { get; set; }
        public string DepartmentTitle { get; set; }
        public string DistrictName { get; set; }
        public string ModifiedByName { get; set; }
        public Nullable<int> ImageAttachmentCount { get; set; }
        public Nullable<int> AttachmentCount { get; set; }
        public string CategoryNameEnglish { get; set; }
        public string CategoryNameHindi { get; set; }
        public string SubCategoryNameEnglish { get; set; }
        public string SubCategoryNameHindi { get; set; }
        public int TotalRecords { get; set; }
        public Nullable<System.DateTime> PressreleaseDate { get; set; }
    }

    public class PressReleaseFilterModel : IndexModel
    {
        public string DepartmentCode { get; set; }
        public string DistrictCode { get; set; }
        public string VIPCategoryCode { get; set; }
        public string VIPPersonCode { get; set; }
        public int? CategoryCode { get; set; }
        public int? SubCategoryCode { get; set; }
        public string ToDate { get; set; }
        public string FromDate { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public int? Status { get; set; }
        public long Id { get; set; }
        public long DIPR_Id { get; set; }
        public string PressReleaseFromDate { get; set; }
        public string PressReleaseToDate { get; set; }
        public int? ModifiedBy { get; set; }

    }

    #region Press release user configration
    public class PressReleaseUserConfigrationModel
    {
        public int Id { get; set; }
        public string UserType { get; set; }
        public Nullable<int> UserId { get; set; }
        public Nullable<long> StartNo { get; set; }
        public Nullable<long> EndNo { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public Nullable<System.DateTime> CreatedDate { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public string UserName { get; set; }
    }
    #endregion
}

File: ANGULAR/ANGULAR/src/app/Shared/Model/TenderPressRelease/press-release-model.ts
import { IndexModel } from "../general-model";

export class PressReleaseModel {
    Id: number;
    Code: number | null;
    DepartmentCode: number | string | null;
    CategoryCode: number | string | null;
    LookupCategoryCode: number | string | null;
    SubCategoryCode: number | string | null;
    DistrictCode: number | string | null;
    GeneralDescription: string;
    Description: string;
    URL: string;
    DisplayOrder: number | null;
    PDFUrl: string;
    ImageUrl: string;
    HomePageImageUrl: string;
    KeyWords: string;
    CreatedBy: number | null;
    ModifiedBy: number | null;
    CreatedDate: Date | string;
    ModifiedDate: Date | string | null;
    IsActive: boolean;
    IsDeleted: boolean;
    VIPCategoryList: string[] | number[];
    VIPPersionList: string[];
    IsSpecialPressRelease: boolean;
    PressReleaseLevelCode: number | string;
    PressreleaseDate: Date | string | null;
    PressReleaseDepartmentMappingList: string[] | number[];
    PDFUrlList: string[] = [];
    DistrictList: string[] = [];
    ImageUrlList: string[] = [];
    IsOldRecord: boolean;
    PressReleaseTime: string;
    DistrictName: string;
    VIPPersonName: string;
    DIPR_Id: number;
    NameOfVIPPerson: string;
    AmountinLakh: string;
    NoOfInaugration: string;
    NoOfLokarpan: string;
    NoOfNewInitatives: string;
    Latitude: string;
    Longitude: string;
}

export class PressReleaseResponseModel {
    Id: number;
    Code: number | null;
    DepartmentCode: number | string | null;
    CategoryCode: number | string | null;
    SubCategoryCode: number | string | null;
    DistrictCode: number | null;
    GeneralDescription: string;
    Description: string;
    URL: string;
    DisplayOrder: number | null;
    PDFUrl: string;
    ImageUrl: string;
    HomePageImageUrl: string;
    KeyWords: string;
    CreatedBy: number | null;
    ModifiedBy: number | null;
    CreatedDate: Date | string;
    ModifiedDate: Date | string | null;
    IsActive: boolean;
    IsDeleted: boolean;
    VIPCategoryList: string;
    VIPPersionList: string[];
}

// export class PressReleaseListModel {
//     Id: number;
//     Description: string;
//     URL: string;
//     PDFUrl: string;
//     ImageUrl: string;
//     HomePageImageUrl: string;
//     IsActive: boolean;
//     DepartmentTitle: string;
//     DistrictName: string;
// }

export class PressReleaseListModel {
    Id: number;
    Description: string;
    URL: string;
    PDFUrl: string;
    ImageUrl: string;
    HomePageImageUrl: string;
    IsActive: boolean;
    DepartmentTitle: string;
    DistrictName: string;
    CreatedDate: Date | string;
    ModifiedDate: Date | string | null;
    DIPR_Id: number | null;
    ModifiedByName: string;
    ImageAttachmentCount: number;
    AttachmentCount: number;
    CategoryNameEnglish: string;
    CategoryNameHindi: string;
    SubCategoryNameEnglish: string;
    SubCategoryNameHindi: string;
}

export class PressReleaseUserConfigrationModel {
    Id: number;
    UserType: string;
    UserId: number | string | null;
    StartNo: number | null;
    EndNo: number | null;
    IsActive: boolean;
    IsDeleted: boolean;
    CreatedDate: Date | string | null;
    CreatedBy: number | null;
    ModifiedDate: Date | string | null;
    ModifiedBy: number | null;
    UserName: string;
}

export class PressReleaseFilterModel extends IndexModel {
    DepartmentCode: string;
    DistrictCode: string;
    VIPCategoryCode: string;
    VIPPersonCode: string;
    CategoryCode: number | null;
    SubCategoryCode: number | null;
    ToDate: string;
    FromDate: string;
    PressReleaseFromDate: string;
    PressReleaseToDate: string;
    IsActive: boolean;
    IsDeleted: boolean;
    Status: number | null;
    Id: number;
    DIPR_Id: number;
    DeptValue: string[];
    DistValue: string[];
    VipPerson: string[];
    VipCategory: string[];
    ModifiedBy: number;

}
